### PR TITLE
Add integration tests for client Native authentication - federation test case

### PR DESCRIPTION
**Pull request by Thumimku**
Feature: https://github.com/wso2/product-is/issues/15684
Test Case is added under is-tests-federation test group.

Following test cases are covered:
1. Create apps and idps.
2. Request Authorisation Code grant via Client Native Authentication  -> Validate Federated Authentication Schema.
3. Perform federated authentication and pass code and state to primary IS via ANA.
4. Validate response contains Authorization Code.

### CODE REVIEW

**Comment by Thumimku** [comment]
> PR builder completed Link: https://github.com/wso2/product-is/actions/runs/7221700298 Status: **failure**

This is because of another integration test failure, not related to this PR. Will trigger again.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/applicationNativeAuthentication/ApplicationNativeAuthenticationFederationTestCase.java [hunk 1]
```diff
@@ -0,0 +1,680 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
```
Comment (pavinduLakshan) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/applicationNativeAuthentication/ApplicationNativeAuthenticationFederationTestCase.java:2]:
```suggestion
 * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
```

Comment (Thumimku) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/applicationNativeAuthentication/ApplicationNativeAuthenticationFederationTestCase.java]: Resolved [cb73fe4](https://github.com/wso2/product-is/pull/18629/commits/cb73fe472876fa56f3ae98730ebce83910178866)

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/applicationNativeAuthentication/Constants.java [hunk 2]
```diff
@@ -49,6 +49,9 @@ public class Constants {
     public static final String REQUIRED_PARAMS = "requiredParams";
     public static final String PROMPT_TYPE = "promptType";
     public static final String PARAMS = "params";
+    public static final String ADDITIONAL_DATA ="additionalData";
+    public static final String REDIRECT_URL ="redirectUrl";
+    public static final String STATE ="state";
```
Comment (janakamarasena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/applicationNativeAuthentication/Constants.java:54]: Formatting

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/applicationNativeAuthentication/ApplicationNativeAuthenticationFederationTestCase.java [hunk 3]
```diff
@@ -0,0 +1,680 @@
...
+    private CloseableHttpClient client;
+    private String primaryISIdpId;
+    private String primaryISAppId;
+    private SCIM2RestClient scim2RestClient;
+    private String secondaryISUserId;
+
+    @DataProvider(name = "configProvider")
+    public static Object[][] configProvider() {
+        return new Object[][]{{TestUserMode.SUPER_TENANT_ADMIN}};
+    }
```
Comment (janakamarasena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/applicationNativeAuthentication/ApplicationNativeAuthenticationFederationTestCase.java:169]: format

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/applicationNativeAuthentication/ApplicationNativeAuthenticationFederationTestCase.java [hunk 4]
```diff
@@ -0,0 +1,680 @@
...
+import static org.wso2.identity.integration.test.applicationNativeAuthentication.Constants.REDIRECT_URL;
+import static org.wso2.identity.integration.test.applicationNativeAuthentication.Constants.REQUIRED_PARAMS;
+import static org.wso2.identity.integration.test.applicationNativeAuthentication.Constants.RESPONSE_MODE;
+import static org.wso2.identity.integration.test.applicationNativeAuthentication.Constants.STATE;
+import static org.wso2.identity.integration.test.applicationNativeAuthentication.Constants.STEP_TYPE;
+import static org.wso2.identity.integration.test.applicationNativeAuthentication.Constants.SUCCESS_COMPLETED;
+import static org.wso2.identity.integration.test.applicationNativeAuthentication.Constants.UTF_8;
+
+/**
+ * Integration test cases for Application Native Authentication federation scenario.
```
Comment (janakamarasena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/applicationNativeAuthentication/ApplicationNativeAuthenticationFederationTestCase.java:115]: Lets update the comment to mention the test case

**Review by janakamarasena — APPROVED** [review]

**Pull request merged**